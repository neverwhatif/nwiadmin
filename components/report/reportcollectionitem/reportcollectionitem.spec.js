import React from 'react';
import { shallow } from 'enzyme';

import ReportCollectionItem from './index';

jest.mock('nwiadmin/components/table', () => 'Table');
jest.mock('../reportcollectionpanel', () => 'ReportCollectionPanel');

const mockColumns = [{
    key: 'lorem',
}, {
    key: 'ipsum',
}];

const mockData = [{
    lorem: 1, ipsum: 2,
}, {
    lorem: 3, ipsum: 4,
}];

describe('ReportCollectionItem', () => {
    const component = shallow(<ReportCollectionItem columns={mockColumns} data={mockData} name="Lorem" />);

    it('should render', () => {
        expect(component).toMatchSnapshot();
    });
});
