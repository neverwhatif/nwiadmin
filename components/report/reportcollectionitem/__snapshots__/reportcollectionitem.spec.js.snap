// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportCollectionItem should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CollectionItem
    columns={
      Array [
        Object {
          "key": "lorem",
        },
        Object {
          "key": "ipsum",
        },
      ]
    }
    data={
      Array [
        Object {
          "ipsum": 2,
          "lorem": 1,
        },
        Object {
          "ipsum": 4,
          "lorem": 3,
        },
      ]
    }
    isDisabled={false}
    name="Lorem"
    type=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "CollectionItem",
      "isDisabled": false,
      "title": "Lorem",
    },
    "ref": null,
    "rendered": "CollectionItem",
    "type": "ReportCollectionPanel",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "CollectionItem",
        "isDisabled": false,
        "title": "Lorem",
      },
      "ref": null,
      "rendered": "CollectionItem",
      "type": "ReportCollectionPanel",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
